CONFIGURATION.md

```diff
@@ -58,14 +58,30 @@ The other placeholders are specified separately.
   # Probe fails if SSL is not present.
   [ fail_if_not_ssl: <boolean> | default = false ]
 
-  # Probe fails if response matches regex.
-  fail_if_matches_regexp:
+  # Probe fails if response body matches regex.
+  fail_if_body_matches_regexp:
     [ - <regex>, ... ]
 
-  # Probe fails if response does not match regex.
-  fail_if_not_matches_regexp:
+  # Probe fails if response body does not match regex.
+  fail_if_body_not_matches_regexp:
     [ - <regex>, ... ]
 
+  # Probe fails if response header matches regex. For headers with multiple values, fails if *at least one* matches
+  fail_if_header_matches:
+    [ - [ header: <string>,
+          [ regex: <regex>, ]
```

Comment:
Indentation seems off here

Reply:
Yeah, does not look quite good. I got it from the prometheus config documentation, e.g. [here](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config)

When the same level contains both optional and non-optional parameters, then it looks like this (comments stripped):

```yml
job_name: <job_name>
[ scrape_interval: <duration> | default = <global_config.scrape_interval> ]
```

In here, though, the `regex` should be named `regexp` and be made non-optional, so it would look like this:

```yml
  # Probe fails if response header matches regex. For headers with multiple values, fails if *at least one* matches.
  fail_if_header_matches:
    [ - [ header: <string>,
          regexp: <regex>,
          [ allow_missing: <boolean> | default = false ]
        ], ...
    ]
```

Would that be OK?

Reply:
Looking around, the only similar one we have is some of pagerduty for the AM. There we break it out as a different type.

Reply:
Another example right in this repo is `tcp_probe`, it does not break it out as a separate type. That's why I did not do it here. Can extract both or keep inlined like above

Reply:
For query_response they're all optional though.

Reply:
OK, extracted only the http header match spec.